tox.ini

```diff
@@ -1,10 +1,50 @@
 [tox]
-envlist = py26,py27,py33,py34,py35,py36,py37,pypy
+envlist = cp{26,27,33,34,35},pp{2,3},cp26-win
+minversion=2.7
 
 [testenv]
-deps = -rtest_requirements.txt
-commands = python -m unittest discover -v
-passenv = LANG LOCALE_ARCHIVE
+recreate = True
+passenv =
+    SHOW_TEST_ENV
+    PYTHONUNBUFFERED
+    LANG
+    LOCALE_ARCHIVE
+basepython =
+    cp26: {env:TOXPY26:python2.6}
+    cp26-win: {env:TOXPY26:python2.6}
+    cp27: {env:TOXPY27:python2.7}
+    cp33: {env:TOXPY33:python3.3}
+    cp34: {env:TOXPY34:python3.4}
+    cp35: {env:TOXPY35:python3.5}
+    cp36: {env:TOXPY36:python3.6}
+    cp37: {env:TOXPY37:python3.7}
+    pp2: {env:TOXPYPY:pypy}
+    pp3: {env:TOXPYPY3:pypy3}
 
-[testenv:py26]
-commands = python -m unittest2.__main__ discover -v
+commands =
+    pip install -rtest_requirements.txt
+    pip --version
+    coverage run --branch -m unittest discover -v -t . haas
+    python -m haas haas
+
+[testenv:cp26-win]
+install_command = pip --trusted-host pypi.org --trusted-host pypi.python.org install {packages}
+commands =
+    pip install -rtest_requirements.txt
```

Comment:
Shouldn't the `-rtest_requirements` be listed with `deps=` instead of it's own command in the test commands?

This is currently the failing point for pypy2.

Reply:
`deps` are installed before the haas package its declared runtime dependences. So installing the test dependencies before installing haas can result in an environment with incompatible package versions. When tox then installs `haas`, the invalid versions are uninstalled and the declared versions in setup.py should be installed. So to avoid unnecessary install/uninstall operations I install the test packages in the `commands` section.